csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -868,6 +941,12 @@ public Settings(bool formatDefaultValues, TypeRegistry typeRegistry) : this(form
             /// </summary>
             /// <param name="formatEnumsAsIntegers"><c>true</c> to format the enums as integers; <c>false</c> to format enums as enum names.</param>
             public Settings WithFormatEnumsAsIntegers(bool formatEnumsAsIntegers) => new Settings(FormatDefaultValues, TypeRegistry, formatEnumsAsIntegers);
+            
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object with the specified indentation and the current settings.
+            /// </summary>
+            /// <param name="indent"></param>
```

Comment:
Please add documentation, including the default value and the use of null/empty string to disable indentation.

Reply:
I think it's also worth documenting that when indentation is enabled, the new-line of the writer becomes relevant (whereas previously it wasn't).

Reply:
> Please add documentation, including the default value and the use of null/empty string to disable indentation.

Sure! Do mean doc comments, or somewhere else? I can't find the C# doc source files anywhere in the repo, on github or even google.

Reply:
I've added extended documentation here in the comments. Also covered the gotcha about `\r\n` on Windows, which causes .NET API to produce only empty lines as Json output.